### PR TITLE
Z coin optional date

### PR DESCRIPTION
**Pull request by smk762**
This PR tests the dev branch API with includes https://github.com/KomodoPlatform/komodo-defi-framework/pull/1967

image

For syncing ARRR, the following cases apply:
- If no prior cache and `use_date` if off, sync starts 2 days in the past.
- If no prior cache and `use_date` if on, sync starts on block height derived from date (rounded to nearest 1000).
- If prior cache exists and `use_date` if off, sync continues from prior cache height.
- If prior cache exists and `use_date` if on, and date value has changed since last sync, sync starts on block height derived from date (rounded to nearest 1000).
- If prior cache exists and `use_date` if on, and date value **has not** changed since last sync, sync continues from prior cache height.

To test:
- log in and activate ARRR to cover the cases above and confirm app performs as expected.

### CODE REVIEW

**Comment by cipig** [comment]
this one really fixes my problem

i have this
image

and after restart it doesn't overwrite the existing DB any more, it continues from where it stopped

**Comment by cipig** [comment]
image

if i just check "use date sync" and keep the date same it starts from
```
21 06:09:48, coins::z_coin::z_rpc:389] INFO Older/Newer sync height detected!, rewinding walletdb to new height: Some(2490000)
21 06:10:05, coins::z_coin::z_rpc:497] INFO Older/Newer sync height detected!, rewinding blocks_db to new height: 2490000
21 06:10:12, coins::z_coin::z_rpc:882] INFO (Re)starting light_wallet_db_sync_loop for ARRR, blocks per iteration 5000, interval in ms 0
```
after restart, overwriting the existing DB

**Review by cipig — DISMISSED** [review]
all 5 cases work as expected
thanks for the fix

**Review by borngraced — APPROVED** [review]

File: CMakeLists.txt [hunk 1]
```diff
@@ -78,7 +78,7 @@ FetchContent_Declare(
 )
 
 FetchContent_Declare(jl777-coins
-        URL https://github.com/KomodoPlatform/coins/archive/master.zip)
+        URL https://github.com/KomodoPlatform/coins/archive/ssl-only-coins-config.zip)
```
Comment (cipig) [inline CMakeLists.txt:81]: this line should be reverted

**Review by cipig — APPROVED** [review]

**Pull request merged**